src/modules/fundingManager/bondingCurveFundingManager/BondingCurveFundingManagerBase.sol

```diff
@@ -0,0 +1,143 @@
+
+    //--------------------------------------------------------------------------
+    // Public Functions
+
+    function token() public view returns (IERC20) {
+        return __Module_orchestrator.token();
+    }
+
+    function buyOrder(uint _depositAmount) external payable virtual {
+        // WiP
+        // Deduct fee from incoming value. Fee is paid in collateral token
+        _issueTokens(_depositAmount, collateral);
+    }
+
+    function openBuy() external onlyOrchestratorOwnerOrManager {
+        // Function to set the PAMM buy functionality to open
+        _openBuy();
+    }
+
+    function closeBuy() external onlyOrchestratorOwnerOrManager {
+        // Function to set the PAMM buy functionality to close
+        _closeBuy();
+    }
+
+    function updateBuyFee(uint _fee) external onlyOrchestratorOwnerOrManager {
+        // Should update the buy fee of the contract
+        _updateBuyFee(_fee);
+    }
+    //--------------------------------------------------------------------------
+    // Public Mutating Functions
+
+    function decimals()
+        public
+        view
+        override(ERC20Upgradeable)
+        returns (uint8)
+    {
+        return tokenDecimals;
+    }
+
+    function deposit(uint amount) external {}
+
+    function depositFor(address to, uint amount) external {}
+
+    function withdraw(uint amount) external {}
+
+    function withdrawTo(address to, uint amount) external {}
+
+    //--------------------------------------------------------------------------
+    // Internal Functions Implemented in Downstream Contract
+    function _issueTokensFormulaWrapper(
+        uint _depositAmount,
+        address _collateral
+    ) internal virtual returns (uint);
+
+    //--------------------------------------------------------------------------
+    // Internal Functions
+
+    function _openBuy() internal {
+        // Validate if not open
+        buyIsOpen = true;
+    }
+
+    function _closeBuy() internal {
+        // Validate if not closed
+        buyIsOpen = false;
+    }
+
+    function _setCollateral(address _collateral) internal {
+        collateral = _collateral;
+    }
+
+    function _updateBuyFee(uint _fee) internal {
+        buyFee = _fee;
+    }
+
+    function _issueTokens(uint _depositAmount, address _collateral)
+        internal
+        returns (uint mintAmount)
+    {
+        mintAmount = _issueTokensFormulaWrapper(_depositAmount, _collateral);
+        _mint(msg.sender, mintAmount);
+    }
+
+    function _setTokenDecimals(uint8 _decimals) internal {
+        // Input check
+        tokenDecimals = _decimals;
+    }
+    //--------------------------------------------------------------------------
+    // OnlyOrchestrator Mutating Functions
+
+    function transferOrchestratorToken(address to, uint amount)
+        external
+        onlyOrchestrator
+    {
+        __Module_orchestrator.token().safeTransfer(to, amount);
```

Comment:
Since we are going to remove orchestrator.token() anyway, maybe go ahead and use the collateral address directly?

Reply:
So the initial idea was to add a collateral address to the contract. The reason I discussed with Felix was because in the future we might want to make use of multiple collateral tokens.

I have however removed the reference to the collateral token in the code I'm currently writing (next PR) because setting up the contract for a future with multiple collateral tokens makes me assume to many things. I decided instead to go with what we need right now which means that the code should work with one collateral token, which is the one defined by the Orchestrator.

For this reason, I would leave the transfer of the collateral like this, and update once we remove the `orchestrator.token()` to however it will be implemented once this is needed.

Happy to hear your thoughts around this? :)

Reply:
Yeah, I think we can keep it like that then. Once we remove `orchestrator.token()` we can refactor it, but not get ahead of ourselves until then. 
Having multiple tokens seems a change big enough to wait for the rest to be in place. Also I think it's out of the scope of the BCRG doc?